tools/dockerbuild/build-push-images.sh

```diff
@@ -136,47 +152,58 @@ echo "Placing binaries in: ${ARTIFACTS}"
 mkdir -p "${ARTIFACTS}"
 gh_endgroup
 
-gh_group "Prepare buildx"
-docker buildx use default
-docker buildx create \
-  --platform linux/amd64,linux/arm64 \
-  --use \
-  --name nessiebuild \
-  --driver-opt network=host || docker buildx use nessiebuild
-# Note: '--driver-opt network=host' is needed to be able to push to a local registry (e.g. localhost:5000)
-gh_endgroup
-
-gh_group "Docker buildx info"
-docker buildx inspect
-gh_endgroup
+if [[ ${LOCAL} == 0 ]] ; then
+  gh_group "Prepare buildx"
+  ${TOOL} buildx use default
+  ${TOOL} buildx create \
+    --platform linux/amd64,linux/arm64 \
+    --use \
+    --name nessiebuild \
+    --driver-opt network=host || docker buildx use nessiebuild
+  # Note: '--driver-opt network=host' is needed to be able to push to a local registry (e.g. localhost:5000)
+  gh_endgroup
+
+  gh_group "Docker buildx info"
+  ${TOOL} buildx inspect
+  gh_endgroup
+fi
 
 #
-# Java multiplatform image
+# Gradle Build
 #
-
-gh_group "Build Java linux/amd64"
+gh_group "Build Java"
 ./gradlew "${GRADLE_PROJECT}:clean" "${GRADLE_PROJECT}:build" -x "${GRADLE_PROJECT}:check"
 gh_endgroup
 
-gh_group "Docker buildx build"
-# All the platforms that are available
-PLATFORMS="linux/amd64,linux/arm64/v8,linux/ppc64le,linux/s390x"
-docker buildx build \
-  -f "${BASE_DIR}/tools/dockerbuild/docker/${DOCKERFILE}" \
-  --platform "${PLATFORMS}" \
-  -t "${IMAGE_NAME}:latest" \
-  -t "${IMAGE_NAME}:latest-java" \
-  -t "${IMAGE_NAME}:${IMAGE_TAG_BASE}" \
-  -t "${IMAGE_NAME}:${IMAGE_TAG_BASE}-java" \
-  "${BASE_DIR}/${PROJECT_DIR}" \
-  --push \
-  --provenance=false --sbom=false \
-  --output type=registry
-  # Note: '--output type=registry' is needed to be able to push to a local registry (e.g. localhost:5000)
-  # Note: '--provenance=false --sbom=false' work around UI issues in ghcr + quay showing 'unknown/unknown' architectures
-  gh_summary "## Java image tags, built for ${PLATFORMS}"
-  gh_summary "* \`docker pull ${IMAGE_NAME}:latest\`"
-  gh_summary "* \`docker pull ${IMAGE_NAME}:latest-java\`"
-  gh_summary "* \`docker pull ${IMAGE_NAME}:${IMAGE_TAG_BASE}\`"
-  gh_summary "* \`docker pull ${IMAGE_NAME}:${IMAGE_TAG_BASE}-java\`"
-gh_endgroup
+if [[ ${LOCAL} == 1 ]] ; then
+  gh_group "Docker build"
+  ${TOOL} build \
+    --file "${BASE_DIR}/tools/dockerbuild/docker/${DOCKERFILE}" \
+    --tag "${IMAGE_NAME}:latest" \
+    --tag "${IMAGE_NAME}:${IMAGE_TAG_BASE}" \
+    "${BASE_DIR}/${PROJECT_DIR}"
+  gh_endgroup
+else
+  gh_group "Docker buildx build"
+  # All the platforms that are available
+  PLATFORMS="linux/amd64,linux/arm64/v8,linux/ppc64le,linux/s390x"
+  ${TOOL} buildx build \
+    --file "${BASE_DIR}/tools/dockerbuild/docker/${DOCKERFILE}" \
+    --platform "${PLATFORMS}" \
+    --tag "${IMAGE_NAME}:latest" \
+    --tag "${IMAGE_NAME}:latest-java" \
+    --tag "${IMAGE_NAME}:${IMAGE_TAG_BASE}" \
+    --tag "${IMAGE_NAME}:${IMAGE_TAG_BASE}-java" \
+    "${BASE_DIR}/${PROJECT_DIR}" \
+    --push \
```

Comment:
Not a full review yet, but usually when I need to build a local image for use in minikube, I don't bother pushing to insecure registries.

Instead I remove this `--push` option and execute `eval $(minikube docker-env)` prior to building the image. This makes the build happen inside minikube, and the image is immediately available in the minikube node.

Have you tried this approach? Is it worth providing support for it as well?

Reply:
Too late :-)

Reply:
The difficulties begin when you use podman :(

Loads of issues with podman in minikube - that was the only approach that worked for me. (Yes, they document that you can use `podman-remote` and let that talk to minikube - but no - a huge API version mismatch... - just to name one annoyance)